### PR TITLE
[ltsmaster] Backport PR #2315.

### PR DESCRIPTION
**Pull request by JohanEngelen**
The change is to fix debuginfo build errors with Phobos and LLVM5.0. This PR is mostly copy-pasting from master and fixing compile errors, so not much checking whether things make sense or not for LTS.

### CODE REVIEW

**Comment by JohanEngelen** [comment]
@redstar As you're busy with lts master, is this PR still valuable? Or should we just drop debuginfo support in lts (because we only use it as bootstrapper)? I reckon debuginfo may help with fixing bugs but...

**Comment by redstar** [comment]
Yes, I consider this PR still valueable. One of my next tasks. I like to have ltsmaster in a good, clean state.

**Comment by redstar** [comment]
@JohanEngelen I do not get the test failure (`running/closure.d`) on ARM. Could you please rebase on ltsmaster?

**Comment by JohanEngelen** [comment]
@redstar done

File: gen/nested.cpp [hunk 1]
```diff
@@ -150,27 +137,42 @@ DValue *DtoNestedVariable(Loc &loc, Type *astype, VarDeclaration *vd,
...
-  val = DtoGEPi(val, 0, idx, vd->toChars());
+  LLValue *gep = DtoGEPi(val, 0, idx, vd->toChars());
+  val = gep;
   IF_LOG {
     Logger::cout() << "Addr: " << *val << '\n';
     Logger::cout() << "of type: " << *val->getType() << '\n';
   }
-  if (byref || (vd->isParameter() && getIrParameter(vd)->arg &&
-                getIrParameter(vd)->arg->byref)) {
+  const bool isRefOrOut = vd->isRef() || vd->isOut();
```
Comment (redstar) [inline gen/nested.cpp]: Evaluation of `isRefOrOut` and `vd->isParameter...` gives different results and as consequence a missing load, leading to the failure in the test suite. I think a simple solution is to add 

    const bool isRefParameter = vd->isParameter() && getIrParameter(vd)->arg && getIrParameter(vd)->arg->byref;

and change the condition below to 

    } else if (byref || isRefOrOut || isRefParameter) {

This makes sure that no load is missed.

Comment (JohanEngelen) [reply gen/nested.cpp]: @redstar you can push to this branch (afaik), can you add your fix?

**Comment by redstar** [comment]
I merged your commit.

**Pull request closed without merging**